src/rhsm/connection.py

```diff
@@ -267,6 +267,13 @@ def _fix_no_proxy():
         log.debug('Environment variable NO_PROXY=%s will be used' % no_proxy)
 
 
+def _get_connection(host, port, context, timeout):
+    if not host.startswith('https') and not host.startswith('HTTPS'):
+        return httplib.HTTPConnection(host, port, timeout=timeout)
```

Comment:
If the host is entered with no protocol at all, e.g. `example.com`, this logic will use an HTTP connection.  I just want to make sure that's what we really want instead of defaulting to HTTPS in all cases except those where HTTP is explicitly requested.  We could lean that way with

```python
if host.startswith('http://') or host.startswith('HTTP://'):
    return httplib.HTTPConnection(...)
else:
  return httplib.HTTPSConnection
```